pkg/cache/tas_flavor_snapshot.go

```diff
@@ -719,6 +826,14 @@ func (s *TASFlavorSnapshot) fillInCounts(requests resources.Requests,
 	}
 }
 
+func belongsToRequiredDomain(leaf *leafDomain, requiredDomain utiltas.TopologyDomainID, levelIdx int) bool {
+	if requiredDomain == "" {
+		return true
+	}
+	// Uses levelValues instead of leaf.id since for topologies with hostname as lowest level it points directly to the hostname
+	return strings.HasPrefix(strings.Join(leaf.levelValues, ","), string(requiredDomain))
```

Comment:
can we use tas.DomainID here? even if we have to cast to string afterwards

Reply:
Do you mean `leaf.id` in this case? If so, then we cannot do it since if the lowest level of topology is hostname the id consists of only hostname value and not all domains in the higher levels of hierarchy

Reply:
Oh yes, sure